src/schemas/generator/v2/index.ts

```diff
@@ -0,0 +1,23 @@
+import { z } from 'zod'
+import { TestRuleSchema } from './rules'
+import { TestDataSchema } from './testData'
+import { TestOptionsSchema } from './testOptions'
+import { ThresholdSchema } from './thresholds'
+
+export const GeneratorFileDataSchema = z.object({
+  version: z.literal('2.0'),
+  recordingPath: z.string(),
+  options: TestOptionsSchema,
+  testData: TestDataSchema,
+  rules: TestRuleSchema.array(),
+  allowlist: z.string().array(),
+  includeStaticAssets: z.boolean(),
+  scriptName: z.string().default('my-script.js'),
+  thresholds: z.array(ThresholdSchema),
```

Comment:
I think you might be able to avoid adding a new schema version if you default thresholds to an empty array:

```ts
  thresholds: z.array(ThresholdSchema).default([])
```

Since it's a new property added to schema, all what `migrate` does is initiate an empty array, which can be done using zod's `default` method.

Reply:
This worked perfectly, thanks for the suggestion! 🙏